### ISSUE TITLE
Jetty client exception on redirect

### ISSUE THREAD

**Issue by rnd4222** [http4s/http4s#2805]
When HTTP server tries to redirect user and `HttpClient.getFollowRedirects() == true`, JettyClient throws an exception on second request:

`java.lang.IllegalStateException: The same org.eclipse.jetty.client.AsyncContentProvider instance cannot be used in multiple requests`

Example code:

```
package test

import cats.effect.{ ExitCode, IO }
import org.eclipse.jetty.client.HttpClient
import org.eclipse.jetty.util.ssl.SslContextFactory
import org.http4s.{ Request, Uri }
import org.http4s.Uri.{ Authority, Scheme }
import org.http4s.client.jetty.JettyClient

object T extends cats.effect.IOApp {
  def run(args: List[String]): IO[ExitCode] = for {
    (c, stop) <- JettyClient.allocate[IO] {
      val c = new HttpClient(new SslContextFactory.Client())
      // uncomment this line to fix
      //c.setFollowRedirects(false)
      c
    }
    req = Request[IO](
      uri = Uri(
        path = "/",
        scheme = Some(Scheme.http),
        authority = Some(Authority(host = Uri.RegName("google.com")))
      )
    )
    res <- c.fetchAs[String](req)
    _ = println(res)
    _ <- stop
  } yield ExitCode.Success
}
```

**Comment by rnd4222** [comment]
I've just figured out that in general it's not possible to repeat `Request` because `Stream` would be closed...

Anyway, is it possible to:

1. Override `followRedirects` on per-request basis when `EntityBody` isn't empty
2. Properly handle repeated requests when body is empty?

Not sure how to check if body is empty though - reading body to see if it is empty can involve side-effects. At least it could be done for **Content-Length** == 0.

**Comment by rossabaker** [comment]
I think we should set that to false by default. We provide that functionality for all backends with the `FollowRedirect` middleware. Would you be interested in submitting a PR to the series/0.20 branch?

**Comment by aeons** [comment]
Closed by #2807
